Python/ast.c

```diff
@@ -4553,10 +4556,12 @@ fstring_find_expr(const char **str, const char *end, int raw, int recurse_lvl,
             /* Start looking for the end of the string. */
             quote_char = ch;
         } else if (ch == '[' || ch == '{' || ch == '(') {
+            if (nested_depth > MAXLEVEL) {
```

Comment:
Unless I am missing something, this needs to be:

```suggestion
             if (nested_depth => MAXLEVEL) {
```

to aboid going out of the bounds of `parenstack`. 

Also, I would add a test that checks what happens if more than MAXLEVEL parenthesis are unbalanced:

```
f'{((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((}'
```

Reply:
Good catch!